### DISCUSSION TITLE
Can AutoFixture obtain instances of xUnit collection fixtures?

### DISCUSSION THREAD

**Discussion by atrauzzi** [AutoFixture/AutoFixture#1329] [Q&A]
I have an xUnit collection fixture that I've successfully configured.  I'd like a way for my test methods to be able to request fixtures from AutoFixture that depend on my xUnit collection fixture.  In order to do this though, I need a way to access it from inside of an AutoFixture auto data attribute.

Is it possible to request xUnit "injections" from inside of a custom auto data attribute?

Alternatively, can I get it from within a customization?

**Comment by aivascu** [comment]
Hello @atrauzzi ,
At this moment AutoFixture does not provide an integration with xUnit's collection fixtures.
I believe requesting injections from inside a data attribute is not possible, since xUnit does not offer this functionality.
AutoFixture's `AutoDataAttribute` is built on top of xUnit's `DataAttribute` (in the case of AutoFixture.XUnit 1 and 2) and has access only what is given by the testing framework. You could try implementing your own `AutoDataAttribute` and get the value via reflection.

I assume you're using a xUnit collection fixture because you want to reuse some instances. AutoFixture's philosophy requires that all tests are self sufficient and do not reuse state, mainly because that might lead to false results and sloppy design. So what I recommend is to create the necessary data inside a customization and then inject it into the (AutoFixture) fixture.
